corehq/apps/domain/views/base.py

```diff
@@ -66,6 +73,36 @@ def select(request, do_not_redirect=False, next_view=None):
         return render(request, domain_select_template, additional_context)
 
 
+Link = namedtuple('Link', ('name', 'url'))
+
+
+@quickcache(['couch_user.username'])
+def get_domain_dropdown_links(couch_user, view_name="domain_homepage"):
+    domains = Domain.active_for_user(couch_user)
+    domain_links = [Link(
+        name=domain_obj.display_name(),
+        url=reverse(view_name, args=[domain_obj.name]),
+    ) for domain_obj in domains]
+
+    linked_domain_links = []
+    domain_names = {d.name for d in domain_links}
+    for domain_obj in domain_links:
+        domain = domain_obj.name
```

Comment:
This code switches between the word `domain` meaning the name and the object (and likewise sometimes using `domain_name` and sometimes using `domain_obj`, such that it is hard to keep track of what type the unadorned word is.

Reply:
I meant to use `domain` for all names and `domain_obj` for all objects. The only inconsistency I see is that `domain_names` could be called `domains` - fixing this in 5eeb7946a9f

Reply:
I think its slightly more confusing than just that one.
Line 81 uses `domains` for a list of objects. and `linked_domains` is also a list of objects in line 94. but the `linked_domain` attribute of `link` is a name, I think.
To add to that confusion, `domain_obj` here isn't actually a domain object, its a `Link` object, unlike the `domain_obj` in line 85, which I believe is an object of class `Domain`.
Anyway, I am not trying to be neurotic about these things. I was just definitely confused going through this code and had to keep referring back to where variables were defined to figure out what type they were.